src/gui_common/charts/line/LineChart.cs

```diff
@@ -0,0 +1,591 @@
+            }
+        }
+
+        // Populate the rows
+        for (int i = 1; i < XAxisTicks; i++)
+        {
+            var label = new Label
+            {
+                SizeFlagsHorizontal = (int)SizeFlags.ExpandFill,
+                Align = Label.AlignEnum.Right,
+            };
+
+            label.Text = Mathf.Round(i * (MaxValues.x - MinValues.x) /
+                (XAxisTicks - 1) + MinValues.x).ToString(CultureInfo.CurrentCulture);
+
+            horizontalLabelsContainer.AddChild(label);
+        }
+
+        // Populate the columns (in reverse order)
+        for (int i = YAxisTicks; i-- > 0;)
+        {
+            var label = new Label
+            {
+                SizeFlagsVertical = (int)SizeFlags.ExpandFill,
+                Align = Label.AlignEnum.Center,
+                Valign = Label.VAlign.Bottom,
+            };
+
+            // Don't set the text for the 0 scale label, since there's already one on the x axis.
+            // Add the label nevertheless so the vertical ticks spacings will remain consistent
+            if (i > 0)
+            {
+                label.Text = Mathf.Round(i * (MaxValues.y - MinValues.y) /
+                    (YAxisTicks - 1) + MinValues.y).ToString(CultureInfo.CurrentCulture);
+            }
+
+            verticalLabelsContainer.AddChild(label);
+        }
+
+        if (!string.IsNullOrEmpty(legendTitle))
+            CreateLegend(legendTitle);
+
+        drawArea.Update();
+    }
+
+    public void ClearChart()
+    {
+        toolTipCallbacks.Clear();
+
+        foreach (var data in DataSets)
+        {
+            ToolTipManager.Instance.ClearToolTips("chartMarkers" + data.Key);
+        }
+
+        foreach (var data in DataSets)
+        {
+            ToolTipManager.Instance.ClearToolTips("chartLegend" + data.Key);
+        }
+
+        // Clear points
+        foreach (Node child in drawArea.GetChildren())
+        {
+            child.QueueFree();
+        }
+
+        // Clear legend
+        foreach (Node child in legendContainer.GetChildren())
+        {
+            child.QueueFree();
+        }
+
+        // Clear abscissas
+        foreach (Node child in horizontalLabelsContainer.GetChildren())
+        {
+            child.QueueFree();
+        }
+
+        // Clear ordinates
+        foreach (Node child in verticalLabelsContainer.GetChildren())
+        {
+            child.QueueFree();
+        }
+    }
+
+    public void UpdateDataSetVisibility(string name, bool visible)
+    {
+        DataSets[name].Draw = visible;
+        drawArea.Update();
+    }
+
+    private void CreateLegend(string title)
+    {
+        // Switch to dropdown if amount of dataset is more than maximum number of icon legends allowed
+        if (DataSets.Count > MaxIconLegend && LegendMode == LegendDisplayMode.Icon)
+            LegendMode = LegendDisplayMode.DropDown;
+
+        switch (LegendMode)
```

Comment:
The cases in this switch are a bit long, I think it should be considered moving them to separate methods.